src/portfft/common/bluestein.hpp

```diff
@@ -0,0 +1,80 @@
+/***************************************************************************
+ *
+ *  Copyright (C) Codeplay Software Ltd.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  Codeplay's portFFT
+ *
+ **************************************************************************/
+
+#ifndef PORTFFT_COMMON_BLUESTEIN_HPP
+#define PORTFFT_COMMON_BLUESTEIN_HPP
+
+#include "portfft/common/host_fft.hpp"
+#include "portfft/defines.hpp"
+
+#include <cmath>
+#include <complex>
+#include <sycl/sycl.hpp>
+
+namespace portfft {
+namespace detail {
+/**
+ * Utility function to get chirp signal and fft
+ * @tparam T Scalar Type
+ * @param ptr Host Pointer containing the load/store modifiers.
+ * @param committed_size original problem size
+ * @param dimension_size padded size
+ */
+template <typename T>
+void get_fft_chirp_signal(T* ptr, std::size_t committed_size, std::size_t dimension_size) {
+  using ctype = std::complex<T>;
+  ctype* chirp_signal = (ctype*)calloc(dimension_size, sizeof(ctype));
+  ctype* chirp_fft = (ctype*)malloc(dimension_size * sizeof(ctype));
```

Comment:
`std::vector`.

Reply:
I typically uses malloc/calloc when I just need an array, and not the full functionality of `std::vector`.
Any reason you suggest using std::vector, or its because of its the C++ way of doing things ?

Reply:
The C++ way of doing things exists to avoid mistakes relating to allocation / frees. It also cuts a couple of lines of code. I don't see the benefit of c-style memory management when we're writing in C++.

Reply:
Started using vectors, in both places